src/test/java/com/teragrep/aer_01/EventContextConsumerTest.java

```diff
@@ -0,0 +1,145 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://github.com/teragrep/teragrep/blob/main/LICENSE>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+
+package com.teragrep.aer_01;
+
+import com.azure.messaging.eventhubs.models.EventContext;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.teragrep.aer_01.config.MetricsConfig;
+import com.teragrep.aer_01.config.source.PropertySource;
+import com.teragrep.aer_01.config.source.Sourceable;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+public class EventContextConsumerTest {
+
+    @Test
+    public void testLatencyMetric() throws Exception {
+        final EventContextFactory eventContextFactory = new CheckpointlessEventContextFactory();
+        final Sourceable configSource = new PropertySource();
+        final int prometheusPort = new MetricsConfig(configSource).prometheusPort;
+        final MetricRegistry metricRegistry = new MetricRegistry();
+
+        try (EventContextConsumer eventContextConsumer = new EventContextConsumer(configSource, new Output.FakeOutput(), metricRegistry, prometheusPort)) {
+            EventContext eventContext = eventContextFactory.create();
+            eventContextConsumer.accept(eventContext);
+            final double records = 10;
+            for (int i = 0; i < records - 1; i++) {
```

Comment:
Maybe `for (int i = 1; i <= records; i++)` just for a bit more clarity. 

And theres also an off by one error here if it should send 10 records:

```
jshell> double records = 10;
records ==> 10.0

jshell> for (int i = 0; i < records - 1; i++) {
   ...>     System.out.println(i);
   ...> }
0
1
2
3
4
5
6
7
8
```